astropy/nddata/nddata.py

```diff
@@ -192,12 +219,51 @@ def mask(self, value):
 
     @property
     def unit(self):
+        """
+        `~astropy.units.Unit`: Unit for the data, if any.
+        """
         return self._unit
 
     @property
     def wcs(self):
+        """
+        any type: wcs for the data, if any.
+
+        Even though nothing is enforced, using `~astropy.wcs.WCS` as ``wcs``
+        attribute is encouraged.
+        """
         return self._wcs
 
     @property
     def meta(self):
+        """
+        `dict`-like: Meta information, if any.
+        """
         return self._meta
+
+    @property
+    def uncertainty(self):
+        """
+        any type: Uncertainty in the data, if any.
+
+        Uncertainty *should* have an attribute ``uncertainty_type`` that is
```

Comment:
This point got a lot of discussion leading up to APE7, the conclusion of which was that uncertainties had to have `uncertainty_type` and that had to be a string. The reasoning was that checking to see if the uncertainty type was `NDUncertainty` was perhaps too restrictive, since one could use some other uncertainty class in principle.